### PR TITLE
Update add license entities

### PR DESCRIPTION
**Pull request by fdianoux**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - bad prettier library configuration, adding semicolon in file needed
+- Update Offer object and Licence object
+
+####License
```
Comment (beInDev) [inline CHANGELOG.md:11]:
```suggestion
#### License
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+      - lastUpdate
+      - number
+  - Update field price, now is an object of type PriceData
+    - PriceData:
+      - priceBandArrowsphereSku
+      - buy_price
+      - sell_price
+      - list_price
+      - currency
+####Offer
```
Comment (beInDev) [inline CHANGELOG.md:31]:
```suggestion
#### Offer
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - bad prettier library configuration, adding semicolon in file needed
+- Update Offer object and Licence object
+
+####License
+- Add more object for define license:
```
Comment (beInDev) [inline CHANGELOG.md:12]:
```suggestion
- Add more objects to define a license:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  - Add object Warning as "warnings" field:
+    - key
+    - message
+  - Add object Config as "configs" field :
+    - name
+    - scope
+    - state
+  - Update field active_seat, now is an object of type ActiveSeatsData:
```
Comment (beInDev) [inline CHANGELOG.md:20]:
```suggestion
  - Update field active_seat, it's now an object of type ActiveSeatsData:
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+    - ActiveSeatsData:
+      - lastUpdate
+      - number
+  - Update field price, now is an object of type PriceData
```
Comment (beInDev) [inline CHANGELOG.md:24]:
```suggestion
  - Update field price, it's now an object of type PriceData
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+    - PriceData:
+      - priceBandArrowsphereSku
+      - buy_price
+      - sell_price
+      - list_price
+      - currency
+####Offer
+- Add more object for define offer:
```
Comment (beInDev) [inline CHANGELOG.md:32]:
```suggestion
- Add more objects to define an offer:
```

File: UPGRADING.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
+# Upgrade guide
+
+## 1.0.3 to 2.0.0
+
+### Changes to the licences client
+
+The object `LicensesClient` now has more information. Object `LicenseOfferFindResult` has been removed,
+it replaced by `OfferFindResult` with new fields, and object `AbstractLicense` extend of `LicenseFindResult`
```
Comment (beInDev) [inline UPGRADING.md:8]:
```suggestion
it's replaced by `OfferFindResult` with new fields, and object `AbstractLicense` extends `LicenseFindResult` 
```

File: UPGRADING.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+            },
+            [OfferFields.COLUMN_ARROW_SUB_CATEGORIES]: [
+                string
+            ],
+        },
+    },
+];
+```
+
+Different type have been replaced :
```
Comment (beInDev) [inline UPGRADING.md:198]:
```suggestion
Different types have been replaced :
```

File: UPGRADING.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+ - `LicenseOfferFindResultData` become `OfferData`
```
Comment (beInDev) [inline UPGRADING.md:200]:
```suggestion
 - `LicenseOfferFindResultData` becomes `OfferData`
```

Same for the 3 next lines

File: package.json [hunk 10]
```diff
@@ -14,7 +14,7 @@
     "doc": "typedoc ./src/index.ts ./src/**/index.ts",
     "lint": "eslint ./{src,tests}/**/*.ts && prettier --list-different ./{src,tests}/**/*.ts",
     "lint:fix": "eslint ./{src,tests}/**/*.ts --fix && prettier --write ./{src,tests}/**/*.ts",
-    "prepublishOnly": "npm run clean && npm run build",
+    "prepare": "npm run clean && npm run build",
```
Comment (beInDev) [inline package.json:17]: Why in prepare instead of prepublish?

**Review by beInDev — CHANGES_REQUESTED** [review]
Nice overhaul 😮 !

Don't forget to check test coverage though, you have two files barely tested (abstractWarning.ts, warningFindResult.ts) and a coverage loss on a third one (abstractLicenseFind.ts)
https://github.com/ArrowSphere/nodejs-api-client/pull/25/checks#step:7:150

No need to do 100% but >90%/file would be nice 👍

File: package.json (hunk 10 above)
Comment (pfeyssaguet) [inline package.json:17]: I'm not sure you can just remove the `prepublishOnly` directive
please check with @beInDev

File: src/licenses/entities/license/activeSeatsFindResult.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { AbstractActiveSeats, ActiveSeatsData } from './abstractActiveSeats';
+
+export class ActiveSeatsFindResult extends AbstractActiveSeats {
```
Comment (pfeyssaguet) [inline src/licenses/entities/license/activeSeatsFindResult.ts:3]: Do we really need an abstract for this?

**Review by pfeyssaguet — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (fdianoux) [reply CHANGELOG.md:11]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (fdianoux) [reply CHANGELOG.md:31]: Done

File: CHANGELOG.md (hunk 3 above)
Comment (fdianoux) [reply CHANGELOG.md:12]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (fdianoux) [reply CHANGELOG.md:20]: Done

File: CHANGELOG.md (hunk 5 above)
Comment (fdianoux) [reply CHANGELOG.md:24]: Done

File: CHANGELOG.md (hunk 6 above)
Comment (fdianoux) [reply CHANGELOG.md:32]: Done

File: UPGRADING.md (hunk 7 above)
Comment (fdianoux) [reply UPGRADING.md:8]: Done

File: UPGRADING.md (hunk 8 above)
Comment (fdianoux) [reply UPGRADING.md:198]: Done

File: UPGRADING.md (hunk 9 above)
Comment (fdianoux) [reply UPGRADING.md:200]: Done

File: package.json (hunk 10 above)
Comment (pfeyssaguet) [reply package.json:17]: Please read this topic: https://stackoverflow.com/questions/17509669/how-to-install-an-npm-package-from-github-directly
TLDR: The goal is to be able to install the dependency in a project without actually publishing a new version. When running `npm i` it still needs to build something and it does so by running the `prepare` script
For the `prepublishOnly` however, I don't know what it's supposed to do so I suggested to @fdianoux to leave both `prepare` and `prepublishOnly` (but maybe you have a better idea)

Comment (beInDev) [reply package.json:17]: I see, but shouldn't we be **wanting** versions and not have a direct unversioned link to GitHub?
This seems unnecessary to me.
If it's for local development capacities npm/yarn link command is more than enough. https://docs.npmjs.com/cli/v8/commands/npm-link

Comment (pfeyssaguet) [reply package.json:17]: what if we want to deploy a project to test it, and not only modifying locally?

File: src/licenses/entities/license/activeSeatsFindResult.ts (hunk 11 above)
Comment (fdianoux) [reply src/licenses/entities/license/activeSeatsFindResult.ts]: Done

File: CHANGELOG.md [hunk 12]
```diff
@@ -6,6 +6,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
```
[the remark this answers is not in the document]
Comment (fdianoux) [reply CHANGELOG.md:6]: Done

File: UPGRADING.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+it replaced by `OfferFindResult` with new fields, and object `AbstractLicense` extend of `LicenseFindResult` 
+has more fields.
+
+v1.0.3:
+
+```js
+[
+    {
+        license: {
+            [LicenseFindResultFields.COLUMN_HIGHLIGHT]: {},
```
[the remark this answers is not in the document]
Comment (fdianoux) [reply UPGRADING.md]: Done

**Review by pfeyssaguet — APPROVED** [review]

File: package.json (hunk 10 above)
Comment (beInDev) [reply package.json:17]: then you can do rc versions I think it would be fine still, but as it will pollute the versions history, it's fine to keep the prepare only and have both git and versions accessible

File: CHANGELOG.md [hunk 14]
```diff
@@ -3,9 +3,55 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.0.0] - 2022-01-05
```
Comment (beInDev) [inline CHANGELOG.md:6]:
```suggestion
## [2.0.0] - 2022-01-06
```

**Review by beInDev — APPROVED** [review]

File: CHANGELOG.md (hunk 14 above)
Comment (fdianoux) [reply CHANGELOG.md]: Done
